compiler/hie-core/.gitignore

```diff
@@ -0,0 +1,2 @@
+dist/
```

Comment:
Might be nice to also add `dist-newstyle` and `cabal.project.local` while you’re at it.